internal/app/machined/internal/platform/baremetal/baremetal.go

```diff
@@ -88,9 +91,27 @@ func (b *BareMetal) Initialize(data *userdata.UserData) (err error) {
 		return err
 	}
 
-	i := installer.NewInstaller(cmdline, data)
-	if err = i.Install(); err != nil {
-		return errors.Wrap(err, "failed to install")
+	// Attempt to discover a previous installation
```

Comment:
This feels like a very heavy pattern to have to follow for every cloud platform. It is only just a copy paste thing, but I was hoping we could come up with something that keeps things simple.

Reply:
This is only implemented for bare-metal like platforms where we need to perform an install inline. VM based cloud providers don't have any of this additional work ( but all do call `owned.MountPointsFromLabels()`).